src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -0,0 +1,317 @@
+      JestClientFactory factory = new JestClientFactory();
+      factory.setHttpClientConfig(new HttpClientConfig.Builder(address)
+              .connTimeout(connTimeout)
+              .readTimeout(readTimeout)
+              .multiThreaded(true)
+              .build()
+      );
+      this.client = factory.getObject();
+      this.version = getServerVersion();
+    } catch (IOException e) {
+      throw new ConnectException(
+              "Couldn't start ElasticsearchSinkTask due to connection error:",
+              e
+      );
+    } catch (ConfigException e) {
+      throw new ConnectException(
+              "Couldn't start ElasticsearchSinkTask due to configuration error:",
+              e
+      );
+    }
+  }
+
+  private Version getServerVersion() throws IOException {
+    // Default to newest version for forward compatibility
+    Version defaultVersion = Version.SIX;
+
+    NodesInfo info = new NodesInfo.Builder().addCleanApiParameter("version").build();
+    JsonObject result = this.client.execute(info).getJsonObject();
+    if (result == null) {
+      LOG.warn("Couldn't get Elasticsearch version");
+      return defaultVersion;
+    }
+
+    JsonObject nodesRoot = result.get("nodes").getAsJsonObject();
+    if (nodesRoot == null || nodesRoot.entrySet().size() == 0) {
+      LOG.warn("Couldn't get Elasticsearch version");
+      return defaultVersion;
+    }
+
+    JsonObject nodeRoot = nodesRoot.entrySet().iterator().next().getValue().getAsJsonObject();
+    if (nodeRoot == null) {
+      LOG.warn("Couldn't get Elasticsearch version");
+      return defaultVersion;
+    }
+
+    String esVersion = nodeRoot.get("version").getAsString();
+    if (esVersion == null) {
+      LOG.warn("Couldn't get Elasticsearch version");
+      return defaultVersion;
+    } else if (esVersion.startsWith("1.")) {
+      return Version.ONE;
+    } else if (esVersion.startsWith("2.")) {
+      return Version.TWO;
+    } else if (esVersion.startsWith("5.")) {
+      return Version.FIVE;
+    } else if (esVersion.startsWith("6.")) {
+      return Version.SIX;
+    }
+    return defaultVersion;
+  }
+
+  public Version getVersion() {
+    return version;
+  }
+
+  private boolean indexExists(String index) {
+    Action action = new IndicesExists.Builder(index).build();
+    try {
+      JestResult result = client.execute(action);
+      return result.isSucceeded();
+    } catch (IOException e) {
+      throw new ConnectException(e);
+    }
+  }
+
+  public void createIndices(Set<String> indices) {
+    for (String index : indices) {
+      if (!indexExists(index)) {
+        CreateIndex createIndex = new CreateIndex.Builder(index).build();
+        try {
+          JestResult result = client.execute(createIndex);
+          if (!result.isSucceeded()) {
+            String msg = result.getErrorMessage() != null ? ": " + result.getErrorMessage() : "";
+            throw new ConnectException("Could not create index '" + index + "'" + msg);
+          }
```

Comment:
Is it possible that two tasks attempt to look for the same index, neither finds it, and so both attempt to create the same index? If so, would it make sense if the create request fails to first check and see if it exists (perhaps again) and to fail only if the index does not already exist?

Reply:
This is the current behavior, but yes, I can add an additional check before throwing the exception.